src/souffle-docker.in

```diff
@@ -0,0 +1,205 @@
+#!/bin/sh
```

Comment:
I don't think that src is appropriate for this file. The docker directory would be better. 

I'm not seeing the advantage to being able to specify a particular commit or github user, either. If the docker containers are set up to run using the current souffle source they would be useful for locally testing, as well as meeting the needs of automated testing. e.g. in the Docker file use soomething like this to set up
`COPY <souffle dir> /souffle`

Reply:
I agree, the docker directory is more appropriate. I originally chose to do things in the way I did as it follows the same convention as the other executable scripts in Souffle, thus making it obvious that it is a script that can be executed, and not just some strange ignorable generated file of which there are a lot. The argument that the docker script should live in the docker directory makes wholly more sense though, so I shall move it there. 

The second point was something I thought about a lot. Using a copy means that one needs both the souffle source and the Dockerfile in order to build a container. Accordingly, the Dockerfiles would then not contain exact build instructions. The purpose of the Dockerfiles is to be self contained units, giving build instructions for a system, without any dependencies except docker itself. Not using a copy also means that anything done in a container is practically entirely reproducible at all times, which I think is especially important in a project where devs are using such a diverse range of systems. Specifying a particular commit and user is useful as one container may be started with the upstream, and another with your code. You can then be sure that they both are built under the same configuration, but cannot effect each other. This has potential use for debugging, I did it that way in the first place because I wanted to check the generated code of the upstream against that of my branch. 

So these might all be good reasons, but it is nonsense to think of a debugging process where a developer makes a change, commits it, builds a docker container, enters that container, executes their command, and repeats that process until the bug is fixed. We want to test against out local code, e.g. by using such a proposed copy. This is where the `run` command comes in, as it mounts the local souffle directory at `/mnt` within the container. This is even better than a copy, as not only can one use their local code, but one can develop on their own system outside the container, with their changes available at all times within the container also.